sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -372,20 +391,21 @@ export class IER12ProcessingService {
    * Check if student has an active restriction.
    * @param studentRestrictions student restrictions
    * @param options check restriction options:
-   * - `restrictionCode`: restriction code.
+   * - `restrictionCodes`: restriction codes.
    * @returns value which indicates
    * if a student has active restriction.
    */
   private checkActiveRestriction(
     studentRestrictions: StudentRestriction[],
-    options?: { restrictionCode?: string },
+    options?: { restrictionCodes?: string[] },
   ): boolean {
     return studentRestrictions?.some(
       (studentRestriction) =>
         studentRestriction.isActive &&
-        (!options?.restrictionCode ||
-          studentRestriction.restriction.restrictionCode ===
-            options.restrictionCode),
+        (!options?.restrictionCodes?.length ||
+          options.restrictionCodes.includes(
+            studentRestriction.restriction.restrictionCode,
+          )),
```

Comment:
Is this code from line 402 to 408 not equivalent to the below?
```ts
 return studentRestrictions?.some(
      ({ isActive, restriction }) =>
        isActive &&
        options?.restrictionCodes?.includes(restriction.restrictionCode),
    );
```

Reply:
I did initially thought about the same, but if we did not mention any options, then the condition will evaluate to false even when there is an active restriction and we need the opposite.